node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdater.java

```diff
@@ -145,11 +161,21 @@ public void converge(State wantedState) {
         currentState = wantedState;
     }
 
-    private void adjustNodeAgentsToRunFromNodeRepository() {
+    void adjustNodeAgentsToRunFromNodeRepository() {
         try {
-            final List<NodeSpec> containersToRun = nodeRepository.getNodes(hostHostname);
-            nodeAdmin.refreshContainersToRun(containersToRun);
-        } catch (Exception e) {
+            Map<String, NodeSpec> nodeSpecByHostname = nodeRepository.getNodes(hostHostname).stream()
+                    .collect(Collectors.toMap(NodeSpec::getHostname, Function.identity()));
+            Map<String, Acl> aclByHostname = Optional.of(cachedAclSupplier.get())
+                    .filter(acls -> acls.keySet().containsAll(nodeSpecByHostname.keySet()))
+                    .orElseGet(cachedAclSupplier::invalidateAndGet);
+
+            Set<NodeAgentContext> nodeAgentContexts = nodeSpecByHostname.keySet().stream()
+                    .map(hostname -> nodeAgentContextFactory.create(
+                            nodeSpecByHostname.get(hostname),
+                            aclByHostname.getOrDefault(hostname, Acl.EMPTY)))
```

Comment:
Asside from a bug in node-repo, the only way this can happen is if a container was de-allocated between the `getNodes()` call and the `getAcls()` call. In that case, using an empty `Acl` is fine, since that container will be removed on next tick anyway.

The reverse case (container allocated between those two calls), the container will be created on the next tick, no different that the current behavior.